src/python/graplctl/graplctl/cli.py

```diff
@@ -675,54 +780,86 @@ def create_dgraph(graplctl_state: GraplctlState, instance_type: str):
 
+    return True
 
-@dgraph.command(help="remove DGraph DNS records")
+
+@dgraph.command(
+    help="spin up a swarm cluster and deploy dgraph on it",
+    name="create",
+)
 @click.option(
-    "-i",
-    "--swarm-id",
-    type=click.STRING,
-    help="unique ID of the swarm cluster",
+    "-t",
+    "--instance-type",
+    type=click.Choice(choices=("i3.large", "i3.xlarge", "i3.2xlarge")),
+    help="EC2 instance type for swarm nodes",
     required=True,
 )
-@click.confirmation_option(
-    prompt="are you sure you want to remove the DGraph DNS records?"
-)
 @pass_graplctl_state
-def remove_dns(graplctl_state: GraplctlState, swarm_id: str) -> None:
-    hosted_zone_id = ROUTE53.list_hosted_zones_by_name(
+def create_dgraph(graplctl_state: GraplctlState, instance_type: str):
+    click.echo(f"creating dgraph cluster of {instance_type} instances")
+    if not _create_dgraph(graplctl_state=graplctl_state, instance_type=instance_type):
+        click.echo("dgraph cluster already exists")
+        return
+    click.echo(f"created dgraph cluster of {instance_type} instances")
+
+
+def _remove_dgraph_dns(graplctl_state: GraplctlState, swarm_id: str):
+    hosted_zone_id = graplctl_state.route53.list_hosted_zones_by_name(
         DNSName=f"{graplctl_state.grapl_deployment_name.lower()}.dgraph.grapl"
     )["HostedZones"][0]["Id"]
     for instance in docker_swarm_ops.swarm_instances(
-        ec2=EC2,
+        ec2=graplctl_state.ec2,
         deployment_name=graplctl_state.grapl_deployment_name,
         version=graplctl_state.grapl_version,
         region=graplctl_state.grapl_region,
         swarm_id=swarm_id,
     ):
-        click.echo(f"removing DNS records for swarm {swarm_id}")
+        click.echo(
+            f"removing dns records for instance {instance.instance_id} swarm {swarm_id}"
+        )
         dgraph_ops.remove_dns_ip(
-            route53=ROUTE53,
+            route53=graplctl_state.route53,
             dns_name=f"{graplctl_state.grapl_deployment_name.lower()}.dgraph.grapl",
             ip_address=instance.private_ip_address,
             hosted_zone_id=hosted_zone_id,
         )
-        click.echo(f"removed DNS records for swarm {swarm_id}")
+        click.echo(
+            f"removed dns records for instance {instance.instance_id} swarm {swarm_id}"
+        )
+
+
+@dgraph.command(help="remove dgraph dns records", name="remove-dns")
+@click.option(
+    "-i",
+    "--swarm-id",
+    type=click.STRING,
+    help="unique id of the swarm cluster",
+    required=True,
+)
+@click.confirmation_option(
+    prompt="are you sure you want to remove the dgraph dns records?"
+)
+@pass_graplctl_state
+def dgraph_remove_dns(graplctl_state: GraplctlState, swarm_id: str):
+    click.echo(f"removing dgraph dns records for swarm {swarm_id}")
+    _remove_dgraph_dns(graplctl_state=graplctl_state, swarm_id=swarm_id)
+    click.echo(f"removed dgraph dns records for swarm {swarm_id}")
```

Comment:
Would it be possible to decompose this module into a few sub-modules, each focused on a coherent subset of graplctl's commands (perhaps at the group level)? Once the file starts getting this size, it starts to become a bit difficult to navigate.

Reply:
Yes, definitely. Will do.

Reply:
Done